src/Processors/QueryPlan/Optimizations/optimizeUseNormalProjection.cpp

```diff
@@ -197,73 +207,120 @@ std::optional<String> optimizeUseNormalProjections(Stack & stack, QueryPlan::Nod
         return true;
     };
 
+    auto projection_query_info = query_info;
+    projection_query_info.prewhere_info = nullptr;
+    if (query.dag)
+        projection_query_info.filter_actions_dag = std::make_unique<ActionsDAG>(query.dag->clone());
+    auto empty_mutations_snapshot = reading->getMutationsSnapshot()->cloneEmpty();
     for (const auto * projection : normal_projections)
     {
         if (!has_all_required_columns(projection))
+        {
+            /// Check if projection can be used to filter parts
+            if (query.filter_node)
+            {
+                filterPartsUsingProjection(
```

Comment:
Can we please add a feature flag for this?

I guess later for projections like
```
    PROJECTION region_proj
    (
        SELECT _part_offset ORDER BY region
    ),
```

we would filter part ranges. So the feature flag can be reused.

Reply:
Sure. Will add `optimize_use_projection_filtering` for this purpose.